internal/cli/atlas_alert_config_create.go

```diff
@@ -0,0 +1,236 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"strings"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"github.com/mongodb/mcli/internal/flags"
+	"github.com/mongodb/mcli/internal/json"
+	"github.com/mongodb/mcli/internal/store"
+	"github.com/mongodb/mcli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+const (
+	datadog  = "DATADOG"
+	slack    = "SLACK"
+	victor   = "VICTOR_OPS"
+	flowdock = "FLOWDOCK"
+	email    = "EMAIL"
+	ops      = "OPS_GENIE"
+	org      = "ORG"
+	pager    = "PAGER_DUTY"
+	sms      = "SMS"
+	group    = "GROUP"
+	user     = "USER"
+)
+
+type atlasAlertConfigCreateOpts struct {
+	*globalOpts
+	event                           string
+	matcherFieldName                string
+	matcherOperator                 string
+	matcherValue                    string
+	metricThresholdMetricName       string
+	metricThresholdOperator         string
+	metricThresholdUnits            string
+	metricThresholdMode             string
+	notificationToken               string // notificationsApiToken, notificationsFlowdockApiToken
+	notificationChannelName         string
+	apiKey                          string // notificationsDatadogApiKey, notificationsOpsGenieApiKey, notificationsVictorOpsApiKey
+	notificationEmailAddress        string
+	notificationFlowName            string
+	notificationMobileNumber        string
+	notificationRegion              string // notificationsOpsGenieRegion, notificationsDatadogRegion
+	notificationOrgName             string
+	notificationServiceKey          string
+	notificationTeamID              string
+	notificationType                string
+	notificationUsername            string
+	notificationVictorOpsRoutingKey string
+	notificationDelayMin            int
+	notificationIntervalMin         int
+	notificationSmsEnabled          bool
+	enabled                         bool
+	notificationEmailEnabled        bool
+	metricThresholdThreshold        float64
+	store                           store.AlertConfigurationCreator
+}
+
+func (opts *atlasAlertConfigCreateOpts) init() error {
+	if opts.ProjectID() == "" {
+		return errMissingProjectID
+	}
+
+	var err error
+	opts.store, err = store.New()
+	return err
+}
+
+func (opts *atlasAlertConfigCreateOpts) Run() error {
+	alert := opts.buildAlertConfiguration()
+	result, err := opts.store.CreateAlertConfiguration(alert)
+
+	if err != nil {
+		return err
+	}
+
+	return json.PrettyPrint(result)
+}
+
+func (opts *atlasAlertConfigCreateOpts) buildAlertConfiguration() *atlas.AlertConfiguration {
+
+	out := new(atlas.AlertConfiguration)
```

Comment:
FYI `out` is only needed when is a parameter, see https://github.com/mongodb/mcli/blob/0f7be0649ffe41a9d782b300fa86f7f0fa7b8390/internal/file/file.go#L17

If is an internal variable it doesn't matter much, no need to do anything here, just letting you know